test_data/makedata_suites/610_analyzers.js

```diff
@@ -0,0 +1,255 @@
+
+        if (obj1Length === obj2Length) {
+            return Object.keys(obj1).every(
+                (key) => obj2.hasOwnProperty(key)
+                   && obj2[key] === obj1[key]);
+        } else {
+          throw new Error(`${analyzer_name} analyzer's type missmatched!`);
+        }
+      };
+
+      //This function will check any analyzer's equality with expected server response
+      function arraysEqual(a, b) {
+        if ((a === b) && (a === null || b === null) && (a.length !== b.length)){
+          throw new Error("Didn't get the expected response from the server!");
+        }
+      }
+
+      // this function will check everything regardin given analyzer
+      function checkAnalyzer(analyzerName, expectedType, expectedProperties, expectedResult, queryResult){
+        if (a.analyzer(analyzerName) === null) {
+          throw new Error(`${analyzerName} analyzer creation failed!`);
+        }
+
+        //checking analyzer's name
+        let testName = a.analyzer(analyzerName).name();
+        let expectedName = `_system::${analyzerName}`;
+        if (testName !== expectedName) {
+          throw new Error(`${analyzerName} analyzer not found`);
+        }
+        progress();
+
+        //checking analyzer's type
+        let testType = a.analyzer(analyzerName).type();
+        if (testType !== expectedType){
+          throw new Error(`${analyzerName} analyzer type missmatched!`);
+        }
+        progress();
+
+        //checking analyzer's properties
+        let testProperties = a.analyzer(analyzerName).properties();
+        checkProperties(analyzerName, testProperties, expectedProperties)
+
+        progress();
+
+        //checking analyzer's query results
+        arraysEqual(expectedResult, queryResult);
+
+        progress();
+      }
+
+      //-------------------------------classifierSingle----------------------------------
+
+      let classifierSingle = `classifierSingle_${dbCount}`;
+      let path = `${PWD}model_cooking.bin`;
+      let classifierSingleType = "classification";
+      let classifierSingleProperties = { 
+        "model_location" : `${path}`, 
+        "top_k" : 1, 
+        "threshold" : 0 
+      };
+      let classifierSingleExpectedResult =[
+        {
+          "all" : [
+            "__label__baking"
+          ]
+        }
+      ];
+
+      let classifierSingleQueryReuslt = db._query(`LET str = "Which baking dish is best to bake a banana bread ?" RETURN {"all": TOKENS(str, "${classifierSingle}")}`);
+
+
+      checkAnalyzer(classifierSingle, classifierSingleType, classifierSingleProperties, classifierSingleExpectedResult, classifierSingleQueryReuslt)
+      
+      //-------------------------------classifierDouble----------------------------------
+
+      let classifierDouble = `classifierDouble_${dbCount}`;
+      let classifierDoubleType = "classification";
+      let classifierDoubleProperties = { 
+        "model_location" : `${path}`, 
+        "top_k" : 2, 
```

Comment:
We're testing "top_k" values for 1, and 2. What about higher values? (5, 10, 20 etc.)

Reply:
We are planning to add more use cases with different top_k values soon.